packages/react-server/src/ReactFlightServer.js

```diff
@@ -3454,9 +3454,14 @@ function enqueueFlush(request: Request): void {
     // happen when we start flowing again
     request.destination !== null
   ) {
-    const destination = request.destination;
     request.flushScheduled = true;
-    scheduleWork(() => flushCompletedChunks(request, destination));
+    scheduleWork(() => {
+      request.flushScheduled = false;
+      const destination = request.destination;
+      if (destination) {
+        flushCompletedChunks(request, destination);
+      }
+    });
```

Comment:
This work uncovered a bug where we didn't properly check whether the destination still existed before attempting to flush.